projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/search-ai-results/search-ai-results.component.ts

```diff
@@ -0,0 +1,152 @@
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Component, OnDestroy, OnInit, ViewEncapsulation } from '@angular/core';
+import { ActivatedRoute, Params } from '@angular/router';
+import { PageComponent, PageLayoutComponent, ToolbarActionComponent, ToolbarComponent } from '@alfresco/aca-shared';
+import { concatMap, finalize, takeUntil } from 'rxjs/operators';
+import { ClipboardService, MaterialModule, ThumbnailService, ToolbarModule } from '@alfresco/adf-core';
+import { AiAnswer, Node } from '@alfresco/js-api';
+import { CommonModule } from '@angular/common';
+import { provideAnimations } from '@angular/platform-browser/animations';
+import { SearchAiInputContainerComponent } from '../search-ai-input-container/search-ai-input-container.component';
+import { TranslateModule } from '@ngx-translate/core';
+import { NodesApiService, SearchAiService } from '@alfresco/adf-content-services';
+import { forkJoin } from 'rxjs';
+
+@Component({
+  standalone: true,
+  imports: [
+    CommonModule,
+    PageLayoutComponent,
+    ToolbarActionComponent,
+    ToolbarModule,
+    MaterialModule,
+    ToolbarComponent,
+    SearchAiInputContainerComponent,
+    TranslateModule
+  ],
+  providers: [provideAnimations()],
+  selector: 'aca-search-ai-results',
+  templateUrl: './search-ai-results.component.html',
+  styleUrls: ['./search-ai-results.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  host: { class: 'aca-search-ai-results' }
+})
+export class SearchAiResultsComponent extends PageComponent implements OnInit, OnDestroy {
+  private _agentId: string;
+  private _error = false;
+  private _loading = true;
+  private _mimeTypeIconsByNodeId: { [key: string]: string } = {};
+  private _nodes: Node[] = [];
+  private restrictionQuery = '';
+  private _searchQuery = '';
+  private _queryAnswer: AiAnswer;
+
+  get agentId(): string {
+    return this._agentId;
+  }
+
+  get error(): boolean {
+    return this._error;
+  }
+
+  get loading(): boolean {
+    return this._loading;
+  }
+
+  get mimeTypeIconsByNodeId(): { [key: string]: string } {
+    return this._mimeTypeIconsByNodeId;
+  }
+
+  get nodes(): Node[] {
+    return this._nodes;
+  }
+
+  get queryAnswer(): AiAnswer {
+    return this._queryAnswer;
+  }
+
+  get searchQuery(): string {
+    return this._searchQuery;
+  }
+
+  constructor(
+    private route: ActivatedRoute,
+    private searchAiService: SearchAiService,
+    private clipboardService: ClipboardService,
+    private thumbnailService: ThumbnailService,
+    private nodesApiService: NodesApiService
+  ) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this.route.queryParams.pipe(takeUntil(this.onDestroy$)).subscribe((params: Params) => {
+      this._agentId = params.agentId;
+      this._searchQuery = params.query ? decodeURIComponent(params.query) : '';
+      this.restrictionQuery = params.restrictionQuery || '';
+      if (this.searchQuery) {
+        this.performAiSearch();
```

Comment:
I changed it a bit. Now I'm exiting earlier from that function. I did not use switchMap here because performAiSearch as subscribe inside so it's not returning Observable to use switchMap. But it should work fine, I tested it and I did not see any issues.